src/main/java/com/gdsc/blog/article/service/ArticleService.java

```diff
@@ -0,0 +1,160 @@
+import com.gdsc.blog.mapper.ArticleMapper;
+import com.gdsc.blog.user.entity.User;
+import com.gdsc.blog.user.entity.UserRole;
+import com.gdsc.blog.user.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class ArticleService {
+
+    private final UserRepository userRepository;
+    private final ArticleRepository articleRepository;
+    private final ArticleMapper articleMapper;
+
+    /**
+     * 게시글 생성
+     *
+     * @param articleCreateDto 게시글 생성 DTO
+     * @param username         작성자
+     */
+    public ArticleDto createArticle(
+            ArticleCreateDto articleCreateDto,
+            String username) {
+
+        // 현재 시간
+        Timestamp timestamp = new Timestamp(System.currentTimeMillis());
+
+        // 유저 정보
+        User user = userRepository.findByUsername(username).orElseThrow();
+
+        // 게시글 생성
+        Article article = Article.builder()
+                .title(articleCreateDto.getTitle())
+                .content(articleCreateDto.getContent())
+                .user(user)
+                .createDate(timestamp)
+                .modifyDate(timestamp)
+                .build();
+
+        return articleMapper.toDto(articleRepository.save(article));
+    }
+
+    /**
+     * 최근 게시글 목록 조회
+     *
+     * @return 게시글 목록
+     */
+    public List<ArticleDto> getRecentArticle() {
+        // get recent article list
+        List<Article> articles = articleRepository.findAll(Sort.by(Sort.Direction.DESC, "createDate")); // 최신순으로 정렬
+
+        // article list to article dto list
+        List<ArticleDto> articleList = new ArrayList<>();
+        articles.forEach(article -> articleList.add(articleMapper.toDto(article)));
+
+        return articleList;
+    }
+
+    /**
+     * 유저 게시글 목록 조회
+     *
+     * @return 게시글 목록
+     */
+    public List<ArticleDto> getUserArticle(User user) {
+        // get user article list
+        List<Article> articles = articleRepository.findByUser(user);
+        log.info("articles: {}", articles);
+
+        // article list to article dto list
+        List<ArticleDto> articleList = new ArrayList<>();
+        articles.forEach(article -> articleList.add(articleMapper.toDto(article)));
+
+        return articleList;
+    }
+
+    /**
+     * id로 게시글 조회
+     *
+     * @param idx 게시글 id
+     * @return Article 객체
+     */
+    public ArticleDto getArticleById(Long idx) { //get article by id
+        Optional<Article> article = articleRepository.findById(idx);
+
+        if (article.isPresent()) {
+            return articleMapper.toDto(article.get());
+        } else {
+            throw new IllegalArgumentException("Not found Article by id");
+        }
```

Comment:
이 부분도 전에 리뷰했던 것처럼 `Optional.orElseThrow()` 메소드를 활용해 주시면 좋을 거 같습니다.

```suggestion
        return articleRepository.findById(idx).orElseThrow(() -> new IllegalArgumentException("Not found Article by ID"));
```